daisy/workflow/test_common.go

```diff
@@ -71,15 +71,15 @@ func init() {
 }
 
 func testWorkflow() *Workflow {
-	ctx, cancel := context.WithCancel(context.Background())
+	cancel := make(chan struct{})
```

Comment:
Why not using the Cancel? I thought cancelling a "parent" context would cascade to "children" contexts, cancelling them as well. In short, how does cancel work?